lib/chef/resource/aws_rds_instance.rb

```diff
@@ -0,0 +1,33 @@
+require 'chef/provisioning/aws_driver/aws_resource'
+
+
+class Chef::Resource::AwsRdsInstance < Chef::Provisioning::AWSDriver::AWSResource
+  aws_sdk_type AWS::RDS::DBInstance
+
+  attribute :db_instance_identifier, kind_of: String, name_attribute: true
+
+  attribute :engine, kind_of: String
+  attribute :engine_version, kind_of: String
+  attribute :db_instance_class, kind_of: String
+  attribute :multi_az, default: false, kind_of: [TrueClass, FalseClass]
+  attribute :allocated_storage, kind_of: Integer, default: 5
+  attribute :iops, kind_of: Integer
+  attribute :publicly_accessible, kind_of: [TrueClass, FalseClass], default: false
+  attribute :master_username, kind_of: String
+  attribute :master_user_password, kind_of: String
+  attribute :db_name, kind_of: String
+  attribute :db_port, kind_of: Integer
+
```

Comment:
The [creates](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#create_db_instance-instance_method) are fairly easy, but its the updates that I am more worried about.

The [`modify_db_instance`](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#modify_db_instance-instance_method) method does not accept tags - they must be managed after creation with [`add_tags_to_resource`](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#add_tags_to_resource-instance_method) and [`remove_tags_from_resource`](http://docs.aws.amazon.com/AWSRubySDK/latest/AWS/RDS/Client.html#remove_tags_from_resource-instance_method).

We have logic in [`aws_provider`](https://github.com/chef/chef-provisioning-aws/blob/3635d8853f6169702c679cbd157a9dc46e2c8860/lib/chef/provisioning/aws_driver/aws_provider.rb#L237-L264) for updating the tags but it only works for certain aws objects.

My idea is to create a module which contains that same basic logic (look at current, diff with desired, add/update/delete as necessary) but has a bunch of setters with defaults so the idea of 'how do I read existing tags' becomes configurable.  Then resource authors would only need to include this module and configure it correctly to gain the benefit of tags.